drivers/hmis_external_apis/app/models/hmis_external_apis/tc_hmis/importers/loaders/custom_assessment_loader.rb

```diff
@@ -29,16 +31,41 @@ def runnable?
 
     protected
 
+    def validate_cde_configs
+      seen_element_ids = Set.new
+
+      required_keys = [:label, :key, :repeats, :field_type]
+      all_keys = (required_keys + [:element_id]).to_set
+      cded_configs.each do |item|
+        # Check for required keys
+        raise "Missing required keys in #{item.inspect}" unless required_keys.all? { |k| item.key?(k) }
+
+        raise "Invalid keys present in #{item.inspect}" unless item.keys.all? { |k| all_keys.include?(k) }
+
+        # If :element_id is present, check for uniqueness
+        element_id = item[:element_id]
+        next unless element_id
+        raise "element_id must be integer in #{item.inspect}" unless element_id.is_a?(Integer)
+        raise "Duplicate element_id: #{element_id}" if seen_element_ids.include?(element_id)
+
+        seen_element_ids.add(element_id)
+      end
+
+      true # Return true if no exceptions were raised, indicating validation passed
+    end
+
     def clobber_records(rows)
       assessment_ids = rows.map do |row|
         row_assessment_id(row)
       end
+      assessment_ids.compact!
 
       scope = model_class.where(data_source_id: data_source.id).where(CustomAssessmentID: assessment_ids)
       scope.preload(:custom_data_elements).find_each do |assessment|
         assessment.custom_data_elements.delete_all # delete should really destroy
         assessment.really_destroy!
       end
+      Hmis::Form::FormProcessor.where(custom_assessment_id: assessment_ids).delete_all
```

Comment:
Oh wait , this isn't right. `assessment_ids` is `CustomAssessmentID`s, right? Then this won't do anything because form processor has a reference to the db it, not the "HUD key" . Not sure why this was needed though, as `really_destroy` on the assessment should delete the form processor?

Reply:
Oh good catch, thanks! Yeah this shouldn't be needed, I'll take it out